### PR TITLE
feat: Notify local endpoints about certificate rotation

### LINKED ISSUE #539: Notify local endpoints when the gateway's id certificate is renewed
So that those endpoints can re-register at their earliest convenience.

This could be implemented as an Android broadcast, for example.

# When it should be triggered

- When the certificate is renewed via the Internet (see: #507).
- When the certificate is renewed via couriers (see: #508).

However it's implemented, it should ideally be generic enough to be reused more generally when the certificate changes for  other reasons (e.g., the user migrated to another public gateway), although triggering the notification for other reasons is outside the scope of this issue.

### PR DESCRIPTION
**Pull request by Filmaluco**
Closes #539

### CODE REVIEW

File: app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package tech.relaycorp.gateway.domain.endpoint
+
+enum class EndpointNotifyAction(val action: String) {
```
[the remark this answers is not in the document]
Comment (Filmaluco) [reply app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt:3]: I guess this one stays (based on the talk you had with @sdsantos) correct?

File: app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    ParcelToReceive("tech.relaycorp.endpoint.NEW_PARCEL"),
```
[the remark this answers is not in the document]
Comment (Filmaluco) [reply app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt:4]: I remember having the intel that for now breaking changes were ok and that's why I did. But yes this is from the deleted file `AwalaAndroid.kt`. I will start the PR for the Ping app ASAP

File: app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Filmaluco) [reply app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt]: Android convention

File: app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Filmaluco) [reply app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt]: need permissions for: relaycorp/relaynet-echo-android

[the remark this answers is not in the document]
Comment (gnarea) [reply app/src/main/java/tech/relaycorp/gateway/domain/endpoint/EndpointNotifyAction.kt]: > I remember having the intel that for now breaking changes were ok and that's why I did

Breaking changes are fine indeed, but if the breaking change also introduces an interoperability issue that breaks production, we should try to do a simultaneous release of the affected components to minimise the disruption.

> need permissions for: relaycorp/relaynet-echo-android

We're not using that anymore so I should probably delete it. Or do you mean `relaynet-ping-android`? I think you do have access to that one.

**Review by gnarea — APPROVED** [review]
Thanks for the changes!

Let's hold off merging this one until https://github.com/relaycorp/awala-endpoint-android/pull/199 is good to go.

**Pull request merged**